### PR TITLE
bump(swift): 5.7.3 to 5.8 and spin off runtime and sdk subpackages

### PR DESCRIPTION
**Pull request by buttaface**
I've finally put together subpackages that split off the Swift runtime libraries and SDKs, so that you can also easily install a Swift SDK for the other architectures and cross-compile, by pulling in `ndk-multilib` for Swift's C dependencies too. You can see an example of using the runtime with the `cookcli` package, which previously had to install the entire Swift toolchain just to use the runtime libraries.

@sylirre and @xtkoba, please review the `scripts/` changes; @Maxython, let me know if it's okay for pacman. I had to put that hack in for `buildorder.py` so it allowed depending on other architectures, and the substitution of `x86-64` for `x86_64` since the linter doesn't allow `_` in package names.

The idea of disabling subpackages if they have no files is that each architecture's runtime/SDK is built on the CI and it only packages that one, unlike QEMU which I guess builds all architectures on every CI run? I checked all the subpackages in this repo first, and every one includes some files and will be built, even if that required placeholders like `ndk-multilib-native-stubs`.

I need to write up some debscripts to put symbolic links for the runtime libraries into `$PREFIX/lib` and then this should be ready, putting the subpackaging and architecture dependency changes up for review in the meantime.

### CODE REVIEW

File: scripts/build/termux_step_setup_variables.sh [hunk 1]
```diff
@@ -76,6 +76,9 @@ termux_step_setup_variables() {
 	export prefix=${TERMUX_PREFIX}
 	export PREFIX=${TERMUX_PREFIX}
 
+	# Explicitly export in case the default was set.
+	export TERMUX_ARCH=${TERMUX_ARCH}
```
Comment (buttaface) [inline scripts/build/termux_step_setup_variables.sh:80]: I noticed this wasn't exported when cross-compiling and an arch wasn't passed in, ie the default arch was used, so I made sure it's always exported here, else `buildorder.py` couldn't find it.

File: scripts/build/termux_create_debian_subpackages.sh [hunk 2]
```diff
@@ -60,6 +60,13 @@ termux_create_debian_subpackages() {
 		[ "$TERMUX_SUBPKG_PLATFORM_INDEPENDENT" = "true" ] && SUB_PKG_ARCH=all
 
 		cd "$SUB_PKG_DIR/massage"
+		# Check that files were actually installed, else don't subpackage.
+		if [ "$(find . -type f -print | head -n1)" = "" ]; then
+			echo "No files in subpackage '$SUB_PKG_NAME' when built for $SUB_PKG_ARCH with package '$TERMUX_PKG_NAME', so"
+			echo "the subpackage was not created. If unexpected, check to make sure the files are where you expect."
+			cd "$TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX"
+			continue
+		fi
```
Comment (xtkoba) [inline scripts/build/termux_create_debian_subpackages.sh:69]: How about only applying to platform-independent subpackages? That should be sufficient if I understand the description correctly.

Comment (xtkoba) [reply scripts/build/termux_create_debian_subpackages.sh:69]: Also it should be possible to restrict to the case when the parent package is platform-*dependent*.

Comment (buttaface) [reply scripts/build/termux_create_debian_subpackages.sh:69]: Those conditions you suggest would be enough to allow these Swift subpackages, but do we generally want to allow empty subpackages? I took this check from [the massage step for regular packages, which enforces the even more stringent requirement that one non-doc file _must_ be installed](https://github.com/termux/termux-packages/blob/33b1397f3699c4ac6f0d3567765571cee00679b0/scripts/build/termux_step_massage.sh#L74), which currently doesn't apply to subpackages, so I put in this softer check that at least one subpackage file must be installed.

I think this is a good check generally for subpackages, let me know what you think.

**Comment by buttaface** [comment]
Kicked off a final CI run, will test those CI artifacts locally, then merge later today.

**Pull request merged**